servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala

```diff
@@ -0,0 +1,162 @@
+package org.http4s
+package servlet
+
+import java.net.InetSocketAddress
+import javax.servlet._
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse, HttpSession}
+
+import cats.effect._
+import cats.implicits.{catsSyntaxEither => _, _}
+import org.http4s.headers.`Transfer-Encoding`
+import org.http4s.server._
+import org.log4s.getLogger
+
+import scala.collection.JavaConverters._
+
+class BlockingHttp4sServlet[F[_]](
+    service: HttpRoutes[F],
+    servletIo: BlockingServletIo[F],
+    serviceErrorHandler: ServiceErrorHandler[F])(implicit F: Effect[F])
+    extends HttpServlet {
+  private[this] val logger = getLogger
+
+  private[this] var serverSoftware: ServerSoftware = _
+
+  // micro-optimization: unwrap the service and call its .run directly
+  private[this] val serviceFn = service.run
```

Comment:
Could be protected and move to abstract class.  Though maybe we save a few nanoseconds going through a field (e.g., `private[this]`) instead of an accessor.